.circleci/config.yml

```diff
@@ -292,15 +302,17 @@ jobs:
           destination: node3/
 
   docker_image_build:
-    <<: *container_config
+    <<: *machine_config
     steps:
       - checkout
-      - *setup_remote_docker
-      - *install_docker_client
       - run:
-          name: Build Docker image
+          name: Build image and smoke test the nodes
           command: |
-            ~/bin/docker build .
+            docker-compose build node1
```

Comment:
I see this PR tests an artefact that is distinct from the ones pushed on Docker Hub. (E.g. does not amend any CI job that builds-and-pushes an image by testing before pushing.) This is ok for https://www.pivotaltracker.com/n/projects/2152594/stories/155124244

Though, because the artefact that the user will fetch when following the documentation is not tested, this does not address https://www.pivotaltracker.com/n/projects/2152594/stories/155124252 (cc @dincho ).

Reply:
I guess if the caching is introduced we can just push the image built by this job